other-docs/guides/code-review/standards.md

```diff
@@ -1,52 +1,206 @@
 
-### PHP specific checks and best practices
-- Require At Least PHP 7.1
-- PHP's timezone should not be changed.
-- Only longhand php tags (`<?php`) should be used. Do not use shorthand tags (`<?`) or legacy tags ASP-style tags (`<%`, `%>`, `<%=`, script tag `<script language="php">`).
-- There is no reference sign on a function call. The reference sign should only be used in function definitions.
-- The backtick operator is not allowed as PHP will attempt to execute the contents as a shell command.
-- When using `in_array()` specify the strict parameter `bool $strict` is also specified.
 
-### Maintain query best practices and check against slow database queries 
-- Queries against post meta can be very slow if queries are performed against `meta_value` or `meta_key`. In addition, user `orderby` in a query has poor performance.
-- To maintain performance, queries should only return less than or equal to a 100 posts.
-- Raw SQL statements should be wrapped in a prepare, `$wpdb->prepare()`.
+## Database
 
-### Require input/output is checked
-- When outputting data from PHP, Use sanitization functions or WordPress output functions.
-- Inputs should be both sanitized and validated to ensure the input is expected.
+### Avoid meta value queries {#meta-queries}
 
-### Protect filesystem's data
-- Do not allow using `__FILE__` in menu slugs.
+*Sniff: `HM.Performance.SlowMetaQuery`, `HM.Performance.SlowOrderBy`*
 
-### Use WordPress core functions when possible
-- Preferred instead of the PHP alternative.
-- Deprecated functions should not be used.
-- Certain WordPress functions and constants have alternatives, prefer the better alternative. For example, use `wp_safe_redirect()` instead of `wp_redirect()`. This validates the redirect is in the whitelist (site by default).
-- Ensure nonce verification is used when making requests to ensure the request is allowed.
+Any `meta_query` that uses the `meta_value` column should not be used. Additionally, ordering results by meta value should not be used.
 
-### Deprecated functions should not be used and use alternative WordPress functions and constants if a better alternative exists
+The meta tables do not have MySQL indexes on the `meta_value` column, so query times can be very long for larger tables. While these may have decent performance with smaller sets of data, the meta tables tend to grow rapidly, and quickly become a significant scaling issue.
 
-### Some PHP functions/operations are not allowed
-- `eval()` is not allowed as it allows execution of arbitrary PHP code and thus a security risk.
-- `create_function()` should not be used, it is deprecated as of PHP 7.2.
-- Do not allow the usage of the `goto` operator, it allows jumping to another section of code which creates for less readable code.
+Consider storing lookup values in the meta key itself. This allows checking for the existence of a meta key (by using `EXISTS` in `meta_compare`) instead, which uses the index directly.
```

Comment:
Just a thought here, do we or should we care if the query is going to go via elastic? We could only tell if `ep_integrate` is explicitly set to true.

Reply:
Yes, that is performanente